### PR TITLE
[iOS] CollectionView minor fixes

### PR DESCRIPTION
**Pull request by adrianknight89**
Fixed a logic error and made a few refactoring attempts

### CODE REVIEW

File: Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewController.cs [hunk 1]
```diff
@@ -64,33 +64,33 @@ public void UpdateLayout(ItemsViewLayout layout)
 
 		protected override void Dispose(bool disposing)
 		{
-			if (!_disposed)
+			if (_disposed)
```
Comment (adrianknight89) [inline Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewController.cs]: reduced nesting here

File: Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewRenderer.cs [hunk 2]
```diff
@@ -170,17 +170,17 @@ NSIndexPath DetermineIndex(ScrollToRequestEventArgs args)
 		void UpdateVerticalScrollBarVisibility()
 		{
 			if (_defaultVerticalScrollVisibility == null)
-				_defaultVerticalScrollVisibility = ItemsViewController.CollectionView.ShowsHorizontalScrollIndicator;
+				_defaultVerticalScrollVisibility = ItemsViewController.CollectionView.ShowsVerticalScrollIndicator;
```
Comment (adrianknight89) [inline Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewRenderer.cs]: Was this on purpose?

File: Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewRenderer.cs [hunk 3]
```diff
@@ -231,11 +231,10 @@ protected override void Dispose(bool disposing)
 
 			if (disposing)
 			{
-				if (Element != null)
-				{
-					TearDownOldElement(Element);
-				}
-				ItemsViewController.Dispose();
+				TearDownOldElement(Element);
```
Comment (adrianknight89) [inline Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewRenderer.cs]: `TearDownOldElement` checks for null case for us.

File: Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewRenderer.cs (hunk 2 above)
Comment (hartez) [reply Xamarin.Forms.Platform.iOS/CollectionView/ItemsViewRenderer.cs]: Looks like it was a bug/typo.

**Pull request merged**